rflx/model/message.py

```diff
@@ -1319,6 +1342,33 @@ def typed_variable(expression: expr.Expr) -> expr.Expr:
         for p in self.paths(FINAL):
             types = {f: t for f, t in self.types.items() if f in self.parameters}
             path = []
+            type_error = False
+
+            for l in p:
+                path.append(l.target)
+                for expression in [l.condition, l.size, l.first]:
+                    if expression == expr.UNDEFINED:
+                        continue
+
+                    error = expression.check_type(rty.Any())
+
+                    self.error.extend(error)
+
+                    if error.check():
+                        self.error.extend(
+                            [
+                                (
+                                    "on path " + " -> ".join(f.name for f in path),
+                                    Subsystem.MODEL,
+                                    Severity.INFO,
+                                    expression.location,
+                                )
+                            ],
+                        )
+                        type_error = True
+
+            if type_error:
+                break
```

Comment:
I suppose this code is for checking if there are any references to undefined fields. Why not just doing a global analysis (over all expressions of all links) instead of looking at each sub-path? It is not interesting to know on which sub-path the undefined field is, if the field is not part of the message.

Reply:
I have tested this. The only downside is that in many error messages the path is no longer reported.

Reply:
I think that is no problem. If a completely undefined field is used, the field is undefined on all possible paths, so that reporting the path brings no benefit. The path is only interesting in cases where a field is undefined only on specific paths (e.g., because the field is optional and thus not part of the path).